modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/field-types/FieldTypeList.es.js

```diff
@@ -96,6 +96,46 @@ const FieldTypeList = ({
 		})
 		.sort(({displayOrder: a}, {displayOrder: b}) => a - b);
 
+	if (document.getElementById('screenReaderSearchResult')) {
+		if (keywords !== '') {
+			if (filteredFieldTypes.length) {
+				document.getElementById(
+					'screenReaderSearchResult'
+				).innerText = sub(
+					Liferay.Language.get(
+						'x-results-returned-for-the-search-term-x'
+					),
+					[filteredFieldTypes.length, keywords]
+				);
+			}
+			else {
+				document.getElementById(
+					'screenReaderSearchResult'
+				).innerText = sub(
+					Liferay.Language.get(
+						`${sub(
+							Liferay.Language.get(
+								'there-are-no-results-for-the-search-term-x'
+							),
+							[keywords]
+						)} ${Liferay.Language.get(
+							'check-your-spelling-or-search-for-a-different-term'
+						)}`
+					),
+					[filteredFieldTypes.length, keywords]
+				);
+			}
+		}
+		else {
+			document.getElementById(
+				'screenReaderSearchResult'
+			).innerText = sub(Liferay.Language.get('search-field-is-empty'), [
+				filteredFieldTypes.length,
+				keywords,
+			]);
+		}
+	}
+
```

Comment:
Getting elements by ID is usually faster than other methods for querying the DOM, but we can still avoid getting the same element twice by "storing" the element we need in a variable before further usages. So, before line 99, I would add:

`const screenReaderSearchResult = document.getElementById('screenReaderSearchResult');`

And then replacing all the subsequent calls to `document.getElementById()` with references to the previously created variable.